### ISSUE TITLE
question on audio features

### ISSUE THREAD

**Issue by hannahgym** [yerfor/GeneFace#117]
given an audio, you extract these features:

'''
     ===== start extract esperanto =====')
    >> f'python data_util/extract_esperanto.py --wav {path} --save_feats' 
    ===== extract deepspeech =====')
  >> f'python data_util/deepspeech_features/extract_ds_features.py --input {path} --output {path.replace(".wav", "_deepspeech.npy")}

   8. calculate audio features
  python data_gen/nerf/extract_hubert_mel_f0.py --video_id=$1
  
  hubert_hidden = get_hubert_from_16k_speech(speech_16k)
  np.save(hubert_npy_name, hubert_hidden.detach().numpy())
  print(f"Hubert extracted at {hubert_npy_name}")
  extract_mel_f0_from_fname(wav_16k_name, out_name=mel_f0_npy_name)
  print(f"Mel and F0 extracted at {mel_f0_npy_name}")
'''

if I am not mistaken, final audio features are computed at step 8; what is not clear to me is that, why you are computing esperanto and deepspeech features? where you are going to use <speranto and deepspeech>? are you using all these extracted audio features: 
- esperanto
- deepspeech
- hubert_npy_name
- mel_f0_npy_name

? 

thanks

**Comment by yerfor** [comment]
Hi, for the latest version of GeneFace, HuBERT and f0 is used as the audio condition. 

For inclusiveness, we extract deepspeech to support AD-NeRF, and extract esperanto to support RAD-NeRF.

**Comment by hannahgym** [comment]
could you plz guide what is happening on[ this function](https://github.com/yerfor/GeneFace/blob/7bf2e65081609ba1de9f2634f657f80161688e5a/data_gen/process_lrs3/process_audio_mel_f0.py#L58).  

what is the main difference between f0 and mel? what is the rule of this delta computation: delta_l = len(mel) - len(f0)

**Comment by yerfor** [comment]
Hi, you can refer to [this TTS repo](https://github.com/NATSpeech/NATSpeech/blob/main/utils/audio/pitch_extractors.py)
